### PR TITLE
stubby: set appdata_dir to a directory under /tmp

### PR DESCRIPTION
**Pull request by jonathanunderwood**
Signed-off-by: Jonathan G. Underwood <jonathan.underwood@gmail.com>

Maintainer: @iamperson347
Compile tested: None
Run tested: mips/Lantiq (Homehub 5a)

Description:
If using stubby for dnssec validation, it needs to store trust
anchor data. This is best stored in memory for performance.
This commit does not enable dnssec validation by stubby.

### CODE REVIEW

**Comment by karlp** [comment]
What did appdata_dir default to?  Does it behave properly if that directotry doesn't exist when it starts?

**Comment by jonathanunderwood** [comment]
> What did appdata_dir default to?

From the docs:

_When the system-level user does have a home directory, stubby will store the for Zero configuration DNSSEC dynamically acquired root trust anchor in a subdirectory called ".getdns" of that home directory. If the system-level user does not have a home directory or the home directory is not writeable or readable, stubby will fallback to the current working directory._

_This can be overruled by supplying a "appdata_dir" in the stubby.yml configuration file. When a "appdata_dir" was specified, that directory will be used for storing data related to Zero configuration DNSSEC immediately, without the other paths being tried._

So, by default, stubby would try to store the keys under `/root/.getdns` (but would not have permission to actually do so).

> Does it behave properly if that directotry doesn't exist when it starts?

Yes; the directory is created by stubby.

**Comment by dibdot** [comment]
@jonathanunderwood  same applies here, probably it's not the best approach to push seven PRs in parallel.

**Comment by jonathanunderwood** [comment]
OK, have pushed a single squashed commit.

File: net/stubby/files/stubby.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@ listen_addresses:
   - 127.0.0.1@5453
   -  0::1@5453
 
+appdata_dir: "/tmp/stubby"
```
Comment (EricLuehrsen) [inline net/stubby/files/stubby.yml]: Following Debian convention, semi-persistent working data like trust anchors should be in `/var/lib/stubby`

Comment (jonathanunderwood) [reply net/stubby/files/stubby.yml]: Yep, makes sense, thanks.

**Comment by jonathanunderwood** [comment]
Closing this PR as it is now included in PR #7122 .

**Pull request closed without merging**